src/Health/HealthController.ts

```diff
@@ -0,0 +1,106 @@
+import BitcoinCore = require('bitcoin-core')
+import { inject, injectable } from 'inversify'
+import { Collection, Db } from 'mongodb'
+import * as Pino from 'pino'
+
+import { childWithFileName } from 'Helpers/Logging'
+
+export const isStatus200 = ({ status }: { status: number }) => status === 200
+
+@injectable()
+export class HealthController {
+  private readonly db: Db
+  private readonly collection: Collection
+  private readonly bitcoinCore: BitcoinCore
+  private readonly logger: Pino.Logger
+
+  constructor(
+    @inject('Logger') logger: Pino.Logger,
+    @inject('DB') db: Db,
+    @inject('BitcoinCore') bitcoinCore: BitcoinCore
+  ) {
+    this.logger = childWithFileName(logger, __filename)
+    this.db = db
+    this.collection = this.db.collection('health')
+    this.bitcoinCore = bitcoinCore
+  }
+
+  async getBlockchainInfo(): Promise<void> {
+    const logger = this.logger.child({ method: 'getBlockchainInfo' })
+
+    const {
+      blocks,
+      verificationprogress,
+      bestblockhash,
+      warnings,
+      size_on_disk,
+    } = await this.bitcoinCore.getBlockchainInfo()
+    const blockchainInfo = { blocks, verificationprogress, bestblockhash, warnings, size_on_disk }
+    logger.trace(
+      {
+        blockchainInfo,
+      },
+      'Blockchain Info retrieved successfully.'
+    )
+    await this.collection.updateOne(
+      { name: 'blockchainInfo' },
+      {
+        $set: {
+          blockchainInfo,
+        },
+      },
+      { upsert: true }
+    )
+  }
+
+  async getWalletInfo(): Promise<void> {
+    const logger = this.logger.child({ method: 'getBalance' })
+
+    const { balance, txcount } = await this.bitcoinCore.getWalletInfo()
+    const walletInfo = { balance, txcount }
+    logger.trace(
+      {
+        walletInfo,
+      },
+      'Bitcoin walletInfo retrieved successfully.'
+    )
+    await this.collection.updateOne(
+      { name: 'walletInfo' },
+      {
+        $set: {
+          walletInfo,
+        },
+      },
+      { upsert: true }
+    )
+  }
+
+  async getNetworkInfo(): Promise<void> {
```

Comment:
refreshNetworkInfo might be a better name.

Also split this method up into multiple private methods. remember each function should do as close to 1 thing as possible.

There can be a private method that only calls and returns the result of `this.bitcoinCore.getNetworkInfo()`. I might call this method getNetworkInfo, hence the better name of this containg method.

Reply:
See the StorageWriter, see if you can make updateNetworkInfo a pipe method something like: 

```ts
refreshNetworkInfo = pipe(
  this.log(LogTypes.trace)('refreshing network info'),
  this.getNetworkInfo,
  this.log(LogTypes.trace)('new info gathered, saving network info'),
  this.updateNetworkInfo,
  this.log(LogTypes.trace)('refreshed network info')
)